src/web/extension.ts

```diff
@@ -2,20 +2,102 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License. See License.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
 
 import * as vscode from "vscode";
+import { dataverseAuthentication } from "./common/AuthenticationProvider";
+import { appTypes, PORTALSURISCHEME, PORTALSWORKSPACENAME } from "./common/Constants";
+import { expBackoff } from "./common/ErrorHandler";
+import { PortalsFS } from "./common/FileSystemProvider";
+import { languageIdtoCodeMap, webpagestowebpagesIdMap, websiteIdtoLanguageMap } from "./common/PortalLanguageProvider";
+import { GetCustomDetails, getfoldername, loadschema } from "./common/PortalSchemaReader";
 
 export function activate(context: vscode.ExtensionContext): void {
-    console.log("Activated web extension!"); // sample code for testing the webExtension
-
+    console.log("Activated Power Portal vscode web extension!"); // sample code for testing the webExtension
+    const portalsFS = new PortalsFS();
+    let initialized = false;
+    let accessToken = '';
     context.subscriptions.push(
         vscode.commands.registerCommand(
             "microsoft-powerapps-portals.webExtension.init",
-            () => {
-                // sample code for testing the webExtension
+            async (args: any) => {
                 vscode.window.showInformationMessage(
-                    "Initializing web extension!"
+                    "Initializing Power Portal vscode web extension!"
                 );
+
+                if (initialized) {
+                    return;
+                }
+                initialized = true;
+
+                if (!args) {
+                    return;
+                }
+                const { appname, entity, entityId, searchParams} = args
+                console.log("extension - " + appname + " organization " + entity + entityId + searchParams);
+
+                const queryParams = new Map<string, string>();
+                for(const p of Object.keys(queryParams))
+                {
+                    if(!p)
+                    queryParams.set(p,queryParams.get(p) as string);
+                    console.log(p + " " + queryParams.get(p) as string)
+                }
+
+                 // load the fetch xml for the specified entity
+                let fetchQueryEntityMap = new Map();
+                ({fetchQueryEntityMap} = loadschema());
+
+                // get portal name and other details to create the base directory
+                const portalsFolderName = expBackoff(getfoldername(fetchQueryEntityMap, entity), {});
```

Comment:
Too many things in a single function. Please make it modular and create other private functions (SRP).